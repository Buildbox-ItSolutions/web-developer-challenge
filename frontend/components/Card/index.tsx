import { FC } from 'react';

import {
    Author,
    Card,
    DeleteIcon,
    DeletePost,
    Message,
    ProfilePicture,
    ProfilePictureWrapper
} from './styles';

interface Props {
    profilePic: string;
    message: string;
    autor: string;
    hendleDelete: (id: string) => void;
    id: string;
}

const CardComponent: FC<Props> = ({ autor, message, profilePic, hendleDelete, id }) => {
    return (
        <Card>
            <DeletePost>
                <DeleteIcon
                    src="/delete.svg"
                    onClick={() => hendleDelete(id)}
                    data-test="card-delete-button"
                />
            </DeletePost>
            <ProfilePictureWrapper>
                <ProfilePicture src={profilePic} />
            </ProfilePictureWrapper>
            <Message data-test="card-message">{message}</Message>
            <Author>
                <small>Enviado por</small>
                <strong data-test="card-author">{autor}</strong>
            </Author>
        </Card>
    );
};

export default CardComponent;
