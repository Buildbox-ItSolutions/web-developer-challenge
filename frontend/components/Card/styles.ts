import styled from 'styled-components';

export const Card = styled.div`
    width: 80%;
    max-height: 525px;
    background-color: #313131;
    border: 1px solid #3b3b3b;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
    padding: 20px;
    grid-column-start: 2;
    grid-column-end: 2;

    @media (min-width: 900px) {
        width: 100%;
        display: grid;
        grid-template-columns: 1fr 2fr;
        grid-template-rows: 1fr auto 2fr;
        grid-template-areas:
            'delete delete'
            'pic message'
            'author author';
    }
`;

export const DeletePost = styled.div`
    grid-area: delete;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: flex-end;
    align-items: center;
`;

export const DeleteIcon = styled.img`
    width: 20px;
    height: 20px;
    cursor: pointer;
`;

export const ProfilePictureWrapper = styled.div`
    width: 100%;
    height: 100%;
    grid-area: pic;
    display: flex;
    align-items: center;
    justify-content: center;
    @media (max-width: 900px) {
        padding: 10px;
    }
`;

export const ProfilePicture = styled.img`
    width: 88px;
    height: 88px;
    border-radius: 36px;
`;

export const Message = styled.p`
    box-sizing: border-box;
    width: 100%;
    height: auto;
    color: #9f9f9f;
    grid-area: message;
    word-break: break-all;
    font-size: 16px;
    display: flex;
    align-items: center;
    padding: 0 30px 20px 0;

    @media (max-width: 900px) {
        padding: 0;
        justify-content: center;
        margin: 10px 0 10px 0;
    }
`;

export const Author = styled.div`
    grid-area: author;
    display: flex;
    flex-direction: column;
    grid-column-start: 2;

    small {
        font-size: 12px;
        color: #5f5f5f;
    }

    strong {
        font-size: 14px;
        color: #7a7a7a;
    }
`;
