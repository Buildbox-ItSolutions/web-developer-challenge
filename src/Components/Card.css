#card {
    width: 100%;
    padding: 2.4rem;
    margin-top: .8rem;

    display: flex;
    align-items: flex-start;
    justify-content: center;
    gap: 3.2rem;

    border-radius: 3px;
    border: 1px solid #3b3b3b;
    background-color: #313131;

    position: relative;
}

#card img {
    width: 8.8rem;
    height: 8.8rem;
    border-radius: 3.6rem;

    margin-top: 5.6rem;
}

#card #info {
    width: 100%;

    margin-top: 5.6rem;
}

#card #info p {
    font-size: 1.6rem;
    line-height: 1.25;
    text-align: left;
    color: #9f9f9f;
}

#card #info #infoNome {
    margin-top: 2.4rem;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
}

#card #info #infoNome > span {
    font-size: 1.2rem;
    font-weight: 500;
    line-height: 1.33;
    letter-spacing: 0.29rem;
    text-align: left;
    text-transform: uppercase;
    color: #5f5f5f;
}

#card #info #infoNome > strong {
    font-size: 1.4rem;
    line-height: 1.29;
    text-align: left;
    color: #7a7a7a;
}

#card button {
    border: none;
    background: none;

    cursor: pointer;

    position: absolute;
    top: 0%;
    right: 2%;

    margin-top: -4rem;
}

#card button > img {
    width: 2rem;
    height: 2rem;
}

@media screen and (max-width: 475px) {
    #card {
        flex-direction: column;
    }

    #card img,
    #card #info {
        margin-top: 0;
    }

    #card button {
        margin-top: 0;
        top: 3%;
    }
}