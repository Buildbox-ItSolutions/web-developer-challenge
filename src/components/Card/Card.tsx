import React from 'react';

import { Container, Props } from '.';

const Card: React.FC<Props> = ({ children, ...rest }) => {
  return <Container {...rest}>{children}</Container>;
};

export default Card;
