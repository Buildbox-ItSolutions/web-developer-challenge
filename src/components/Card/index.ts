export { default } from './Card'
export * from './Card.interface'
export * from './Card.styles'
