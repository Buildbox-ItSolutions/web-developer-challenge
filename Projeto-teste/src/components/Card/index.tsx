import { useState } from 'react';
import { BiTrashAlt } from 'react-icons/bi';
import * as Styled from './styles';

interface Props {
  newImage: string;
  setNewImage: Blob | MediaSource;
  newName: string;
  newMessage: string;
  handleImage?: React.ChangeEventHandler;
  handleName?: React.ChangeEventHandler;
  handleMensagem?: React.ChangeEventHandler;
  handleClear: React.MouseEventHandler;
  handleSubmit?: React.FormEventHandler;
  removeImage: React.MouseEventHandler;
}

function Index({
  newImage,
  newName,
  newMessage,
  setNewImage,
  handleImage,
  handleName,
  handleMensagem,
  handleClear,
  handleSubmit,
  removeImage,
}: Props) {
  const [endImage] = useState<string>('logouser.svg');

  return (
    <Styled.Card>
      <Styled.InputImg type="file" onChange={handleImage} value={newImage} />
      <Styled.ImgWrapper>
        {setNewImage ? (
          <>
            <Styled.UserImg
              src={URL.createObjectURL(setNewImage)}
              alt="image"
            />
            <Styled.IconTrash>
              <BiTrashAlt onClick={removeImage} />
            </Styled.IconTrash>
          </>
        ) : (
          <Styled.UserImgPrev src={endImage} alt="image2" />
        )}
      </Styled.ImgWrapper>

      <Styled.InputUser
        onChange={handleName}
        value={newName}
        type="text"
        placeholder="Digite seu nome"
      />
      <Styled.InputMessage
        onChange={handleMensagem}
        value={newMessage}
        placeholder="Menssagem"
      />
      <Styled.GridButton>
        <Styled.ButtonClear onClick={handleClear}>Descartar</Styled.ButtonClear>
        <Styled.ButtonPost onSubmit={handleSubmit}>Publicar</Styled.ButtonPost>
      </Styled.GridButton>
    </Styled.Card>
  );
}

export default Index;
