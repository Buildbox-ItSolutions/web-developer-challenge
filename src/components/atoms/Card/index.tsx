import React from 'react';
import styled from 'styled-components';
import { ThemeSpacing } from 'styles/theme';

type HTMLDivType = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

export interface CardProps extends HTMLDivType {
  paddingLeft?: ThemeSpacing;
  paddingRight?: ThemeSpacing;
  paddingTop?: ThemeSpacing;
  paddingBottom?: ThemeSpacing;
  marginLeft?: ThemeSpacing;
  marginRight?: ThemeSpacing;
  marginTop?: ThemeSpacing;
  marginBottom?: ThemeSpacing;
  minHeight?: HTMLDivType['style']['minHeight'];
}

const Card = styled.div<CardProps>`
  flex: 1;
  border-radius: 3;

  padding-left: ${props => props.theme.spacing[props.paddingLeft]};
  padding-right: ${props => props.theme.spacing[props.paddingRight]};
  padding-top: ${props => props.theme.spacing[props.paddingTop]};
  padding-bottom: ${props => props.theme.spacing[props.paddingBottom]};
  margin-left: ${props => props.theme.spacing[props.marginLeft]};
  margin-right: ${props => props.theme.spacing[props.marginRight]};
  margin-top: ${props => props.theme.spacing[props.marginTop]};
  margin-bottom: ${props => props.theme.spacing[props.marginBottom]};

  border: solid 1px ${props => props.theme.colors.gray_scale_4};
  background-color: ${props => props.theme.colors.gray_scale_2};
  min-height: ${props => props.minHeight};
`;

Card.defaultProps = {
  paddingBottom: '7',
  paddingTop: '7',
  paddingLeft: '7',
  paddingRight: '7',
  marginBottom: '0',
  marginTop: '0',
  marginLeft: '0',
  marginRight: '0'
};

export default Card;
