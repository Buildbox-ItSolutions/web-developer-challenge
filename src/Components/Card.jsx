import del from "../assets/delete.svg"

import "./Card.css"

function Card({img, nome, mensagem, handleClick}) {
    return (
        <div id="card">
            <img src={img} alt="foto de pefil" />

            <div id="info">
                <p>{mensagem}</p>
                
                <div id="infoNome">
                    <span>enviado por</span>
                    <strong>{nome}</strong>
                </div>
            </div>

            <button onClick={handleClick}>
                <img src={del} alt="butÃ£o para deletar a postagem" />
            </button>
        </div>
    )
}

export default Card