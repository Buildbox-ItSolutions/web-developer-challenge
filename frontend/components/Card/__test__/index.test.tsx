import 'jest-styled-components';

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../index';

const props = {
    profilePic: '/default-profile.png',
    message: 'hello',
    autor: 'Hudson',
    id: '1234'
};

describe('Card', () => {
    it('must render without errors', () => {
        const hendleDelete = jest.fn();
        const wrapper = renderer.create(<Card hendleDelete={hendleDelete} {...props} />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('should call handleDelete', () => {
        const hendleDelete = jest.fn();
        const wrapper = shallow(<Card hendleDelete={hendleDelete} {...props} />);

        wrapper.find(`[data-test='card-delete-button']`).simulate('click');

        expect(hendleDelete).toHaveBeenCalled();
    });

    it('should display the author and the message', () => {
        const hendleDelete = jest.fn();
        const wrapper = shallow(<Card hendleDelete={hendleDelete} {...props} />);

        expect(wrapper.find(`[data-test='card-author']`).text()).toEqual('Hudson');
        expect(wrapper.find(`[data-test='card-message']`).text()).toEqual('hello');
    });
});
