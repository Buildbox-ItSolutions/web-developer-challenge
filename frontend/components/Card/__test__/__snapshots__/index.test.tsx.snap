// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card must render without errors 1`] = `
.c0 {
  width: 80%;
  max-height: 525px;
  background-color: #313131;
  border: 1px solid #3b3b3b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 20px;
  padding: 20px;
  grid-column-start: 2;
  grid-column-end: 2;
}

.c1 {
  grid-area: delete;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  width: 20px;
  height: 20px;
  cursor: pointer;
}

.c3 {
  width: 100%;
  height: 100%;
  grid-area: pic;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 88px;
  height: 88px;
  border-radius: 36px;
}

.c5 {
  box-sizing: border-box;
  width: 100%;
  height: auto;
  color: #9f9f9f;
  grid-area: message;
  word-break: break-all;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 30px 20px 0;
}

.c6 {
  grid-area: author;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  grid-column-start: 2;
}

.c6 small {
  font-size: 12px;
  color: #5f5f5f;
}

.c6 strong {
  font-size: 14px;
  color: #7a7a7a;
}

@media (min-width:900px) {
  .c0 {
    width: 100%;
    display: grid;
    grid-template-columns: 1fr 2fr;
    grid-template-rows: 1fr auto 2fr;
    grid-template-areas: 'delete delete' 'pic message' 'author author';
  }
}

@media (max-width:900px) {
  .c3 {
    padding: 10px;
  }
}

@media (max-width:900px) {
  .c5 {
    padding: 0;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin: 10px 0 10px 0;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      data-test="card-delete-button"
      onClick={[Function]}
      src="/delete.svg"
    />
  </div>
  <div
    className="c3"
  >
    <img
      className="c4"
      src="/default-profile.png"
    />
  </div>
  <p
    className="c5"
    data-test="card-message"
  >
    hello
  </p>
  <div
    className="c6"
  >
    <small>
      Enviado por
    </small>
    <strong
      data-test="card-author"
    >
      Hudson
    </strong>
  </div>
</div>
`;
